src/renderers/webgl/WebGLUniformsGroups.js

```diff
@@ -258,41 +259,32 @@ function WebGLUniformsGroups( gl, info, capabilities, state ) {
 
 			}
 
-			// the following two properties will be used for partial buffer updates
-
-			uniform.__data = new Float32Array( infos.storage / Float32Array.BYTES_PER_ELEMENT );
-			uniform.__offset = offset;
-
-			//
+			// add padding to offset
 
 			if ( i > 0 ) {
 
-				chunkOffset = offset % chunkSize;
-
-				const remainingSizeInChunk = chunkSize - chunkOffset;
-
-				// check for chunk overflow
+				const remainingSizeInChunk = euclideanModulo( - offset, chunkSize );
 
-				if ( chunkOffset !== 0 && ( remainingSizeInChunk - infos.boundary ) < 0 ) {
+				if ( remainingSizeInChunk < infos.boundary ) {
```

Comment:
@mrdoob @Mugen87 What this check (checking that `remainingSizeInChunk` is less than `infos.boundary`) does?
Without it we could clean up this code even more (moving the calculation of padded offset just under the `offset += infos.storage` line and removing `i > 0` check and the repad after the loop).

Reply:
I remember it was necessary to implement this check to avoid an overflow issue. However, I'm not sure we have an example so far that fails when the check is removed.

Reply:
OK, let's keep it as it is then.